core/src/test/java/org/apache/iceberg/TestRewriteFiles.java

```diff
@@ -612,33 +612,4 @@ public void testAlreadyDeletedFile() {
     Assert.assertEquals("Only 3 manifests should exist", 3, listManifestFiles().size());
   }
 
-  @Test
-  public void testNewDeleteFile() {
```

Comment:
Why are you removing tests? This is probably incorrect.

Reply:
I see that this is related to removing the validation in `BaseRewriteFiles`. Like I noted there, we want to keep that validation and use it in certain cases.